easybuild/tools/filetools.py

```diff
@@ -335,10 +278,10 @@ def search_file(paths, query, short=False, ignore_dirs=None, silent=False):
     """
     Search for a particular file (only prints)
     """
-    if ignore_dirs is None:
-        ignore_dirs = ['.git', '.svn']
-    if not isinstance(ignore_dirs, list):
-        _log.error("search_file: ignore_dirs (%s) should be of type list, not %s" % (ignore_dirs, type(ignore_dirs)))
+    if not ignore_dirs:
+        ignore_dirs = build_option('ignore_dirs')
+    assert isinstance(ignore_dirs, list), \
+        "search_file: ignore_dirs (%s) should be of type list, not %s" % (ignore_dirs, type(ignore_dirs))
```

Comment:
these are not things that should happen all of a sudden during normal use of easybuild, but they are programming errors, they will show up when a programmer makes a mistake, I think they are very suitable here, see https://wiki.python.org/moin/UsingAssertionsEffectively

They make the amount of jumps, and the overall complexity of this method a bit less severe.